content/blog/2024/12/how-to-migrate-from-idf-bootloader-to-mcuboot-using-esp-self-reflasher/index.md

```diff
@@ -0,0 +1,258 @@
+
+### Building the reflashing images (MCUboot and NuttX application)
+
+Assuming that the NuttX workspace is set, prepare the NuttX application that will be the final target of this process:
+
+The application that will be used is the NSH configured for MCUboot compatibility:
+
+```bash
+cd <NUTTX_DIR>
+./tools/configure.sh esp32-devkitc:mcuboot_nsh
+```
+
+However, it is possible to manually configure the example for building it with MCUboot compatibility. Enter the menuconfig:
+
+```bash
+make menuconfig
+```
+
+Enter the menu `System Type -> Bootloader and Image Configuration` and enable the `Enable MCUboot-bootable format` option.
+
+Ensure that the organization of image slots in flash is properly configured on the menu `System Type -> SPI Flash Configuration`. For this example using ESP32, the following is considered (note that the Bootloader offset may differ among other Espressif chip families):
+
+| REGION                           | OFFSET    | SIZE     |
+| -------------------------------- | --------- | -------- |
+| Bootloader                       | 0x001000  | 0x00F000 |
+| Application primary slot         | 0x010000  | 0x100000 |
+| Application secondary slot       | 0x110000  | 0x100000 |
+| Scratch                          | 0x210000  | 0x040000 |
+
+---
+If MCUboot is being build separately from NuttX build, be sure that the configuration for flash organization matches on both.
+
+---
+
+Save and exit the menuconfig.
+
+Build the MCUboot bootloader:
+
+```bash
+make -j bootloader
+```
+
+Now build the NuttX application:
+
+```bash
+make ESPTOOL_BINDIR=./ -s -j
+```
+
+### Building the Reflashing application
+
+#### Example 1: Download the Target Reflashing Image
+
+ESP Self-Reflasher component provides the example `boot_swap_download_example`, this can be used as start point for the Reflashing application.
+
+```bash
+cd <IDF_DIR>/components/esp-self-reflasher/examples/boot_swap_download_example/
+```
+
+Firstly, copy the final target images to the directory that will be served for HTTP download.
+
+```bash
+cp <NUTTX_DIR>/nuttx.bin <IDF_DIR>/components/esp-self-reflasher/examples/boot_swap_download_example/example_bin_dir/app_upd.bin
+cp <NUTTX_DIR>/mcuboot-esp32.bin <IDF_DIR>/components/esp-self-reflasher/examples/boot_swap_download_example/example_bin_dir/mcuboot-esp32.bin
+```
+
+Configure the example:
+
+```bash
+idf.py menuconfig
+```
+
+Enter menu `Example Configuration -> Bootloader reflash configuration` and set the `Bootloader bin url endpoint` with the server IP, port and the bootloader binary name (in this example, "mcuboot-esp32.bin"):
+
+```
+http://<SERVER_IP>:<PORT>/mcuboot-esp32.bin
+```
+
+Also configure the `Bootloader region address` (the Bootloader offset mentioned above) and the `Bootloader region size`.
+
+Then configure the similar parameters on the `Example Configuration -> Application reflash configuration` menu, set the `Application upgrade url endpoint`:
+
+```
+http://<SERVER_IP>:<PORT>/app_upd.bin
```

Comment:
What should be the IP and port? Is it the user computer IP?

Reply:
Changed it to HOST_IP